src/chatd.cpp

```diff
@@ -2088,6 +2109,12 @@ void Chat::handleTruncate(const Message& msg, Idx idx)
                 CALL_DB(setLastReceived, 0);
             }
         }
+
+        if (mClient.isMessageReceivedConfirmationActive() && mLastIdxReceivedFromServer <= idx)
```

Comment:
You seem to duplicate mLastReceivedIdx and mLastReceivedId. Is there a reason for that?